packages/widget-react/README.md

```diff
@@ -182,6 +192,73 @@ function App() {
 }
 ```
 
+#### Custom Identity Provider
+
+In order to make Custom Identity Provider work, you'll have to properly implement and provide a set of following methods:
+
+- `getToken` - resolving [Chat Widget token]('https://developers.livechat.com/docs/extending-chat-widget/custom-identity-provider#chat-widget-token'). If you want to cache the token, this should return the cached token instead of a fresh request to https://accounts.livechat.com/customer/token endpoint.
+- `getFreshToken` - resolving [Chat Widget token]('https://developers.livechat.com/docs/extending-chat-widget/custom-identity-provider#chat-widget-token'). This should always make a call for a fresh token from https://accounts.livechat.com/customer/token endpoint.
+- `hasToken` - resolving boolean. It determines whether a token has been acquired.
+- `invalidate` - resolving nothing. When called, it should remove the current token. There is no need to do anything else as a new token will be requested by getFreshToken afterwards.
+
+##### Example usage
+
+```ts
+import { LiveChatWidget } from '@livechat/widget-react'
+
+const customIdentityProvider = () => {
+  const baseAPI = 'YOUR_API_URL'
+  const userId = '30317220-c72d-11ed-2137-0242ac120002'
+
+  const getToken = async () => {
```

Comment:
Those function examples don't seem to be really helpful as all they do is some fetch to some imaginary url and response.json 🤔

Reply:
IMO it's the simplest way to achieve that information, and that's how we should approach the subject in terms of README.